uwrt_mars_rover_autonomy/spiralSearch/src/spiralSearch_server.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Action server for the spiral search algorithm
+Currently works with turtlesim for testing 
+*/
+
+#include "ros/ros.h"
+#include <actionlib/server/simple_action_server.h>
+#include <spiralSearch/spiralSearchAction.h>
+#include "geometry_msgs/Twist.h" // contains the ros message type Twist
+#include "turtlesim/Pose.h" // this allows the node read the pose message type published by the node
+#include <sstream> 
+#include <cmath>
+
+//Action Server variables 
+typedef actionlib::SimpleActionServer<spiralSearch::spiralSearchAction> Server;
+spiralSearch::spiralSearchFeedback feedback; 
+spiralSearch::spiralSearchResult result; 
+
+const double PI = 3.14159265359;
+
+//To publish commands to turtlesim
+ros::Publisher velocity_publisher;
+ros::Subscriber pose_subscriber;	
+turtlesim::Pose turtlesim_pose;
+
+//Helper callback function for the turtlesim 
+void poseCallback(const turtlesim::Pose::ConstPtr & pose_message){
+    turtlesim_pose.x = pose_message->x;
+    turtlesim_pose.y = pose_message->y;
+    turtlesim_pose.theta = pose_message->theta;
+}
+
+//Spiral Search Algorithm is implemeted here
+void execute(const spiralSearch::spiralSearchGoalConstPtr &goal, Server* as){
+    
+    //Main spiral search algorithm
+    //declare vel_msg as a "Twist" type to be able to change linear and angular velocities
+    geometry_msgs::Twist vel_msg;
+
+    double angular_velocity = 5; //this is the constant angular velocity, it controls the speed of rover as it moves around
```

Comment:
these constants should be passed in as ros params through the launch file. This makes it easy to configure without going through the code.